src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs

```diff
@@ -130,17 +132,32 @@ impl InputFormatTextBase for InputFormatCSV {
             .expect("must success");
         for (i, end) in batch.row_ends.iter().enumerate() {
             let buf = &batch.data[start..*end];
-            Self::read_row(
+            if let Err(e) = Self::read_row(
                 field_decoder,
                 buf,
                 columns,
                 &builder.ctx.schema,
                 &batch.field_ends[field_end_idx..field_end_idx + n_column],
                 &batch.path,
                 start_row + i,
-            )?;
+            ) {
+                if builder.ctx.on_error_mode == OnErrorMode::Continue {
+                    columns.iter_mut().for_each(|c| {
+                        // check if parts of columns inserted data, if so, pop it.
+                        if c.len() > num_rows {
+                            c.pop_data_value().expect("must success");
+                        }
+                    });
+                    start = *end;
+                    field_end_idx += n_column;
+                    continue;
```

Comment:
if we want to abandon all remaining data of a file

1. abort is the default behavior, nothing to do.
1. continue should be done at an upper level? processor?
2. processer is processing a  ReadBatch/RowBatch of a file,  may need a mutable shared ctx to record the failure of a file.

Reply:
Structured data files don't need to drop remaining data when encounter error.

Reply:
```
In Snowflake, when reading JSON encounter an error on continue mode, they pop the wrong row, and skip all remaining data
```

Reply:
#9336